assets/js/timetree.js

```diff
@@ -1,187 +1,261 @@
+          links.push({
+            source: index,
+            target: branchIndex[branchId],
+            value: 1
+          })
+        }
+      });
 
-        // subset the available angle range; branches go from + to - angle adjust,
-        // so angle adjustment times two divided by number of children minus 1
-        let subangle_adjust = (angle_adjust * 2) / (n_children - 1);
+      TreeGraph({nodes: nodes, links: links, centuries: centuries});
 
-        // start at full negative angle adjustment
-        let current_angle_adjustment = - angle_adjust;
-        for (let i = 0; i < n_children; i++) {
-          // draw a line for each requested sub branch
+   });
 
-          // randomize the number of child branches
-          let num_children = Math.max(2, Math.floor(Math.random() * 7) + 1);
 
-          // drawline(x, y, length * length_ratio, Math.max(2, n_children - 1), angle + current_angle_adjustment, width/2, depth + 1);
-          drawline(x, y, length * length_ratio, num_children, angle + current_angle_adjustment, width/2, depth + 1);
-          // adjust the angle for the next branch
-          current_angle_adjustment += subangle_adjust;
-        }
-      }
+function TreeGraph({nodes, links, centuries}) {
+  let width = 1000;
+  let height = 300;
+
+  let min_x = -width / 2;
+  let min_y = -height / 2;
+
+  let svg =  d3.select("main")
+      .append("svg")
+      .attr("viewBox", [min_x, min_y, width, height]);
+
+  // create a section for the background
+  let background = svg.append("g")
+    .attr("id", "background");
+
+  // create containers for the leaves by century
+  const leafContainerHeight = 45;
+  const leafContainers = background.selectAll("rect")
+      .data(centuries)
+      .join("rect")
+        .attr("id", d => "c" + d)
+        .attr("height", leafContainerHeight)
+        .attr("width", width)
+        .attr("x", min_x)
+        .attr("y", (d, i) => (min_y + i * leafContainerHeight))
+        .attr('fill', "lightgray")
+        .attr('fill-opacity', 0.1)
+        .attr('stroke', 'gray')
+        .attr('stroke-opacity', 0.5)
+        .attr('stroke-dasharray', 2);
+
+
+  // create labels for the centuries
+  const leafContainerLabels = background.selectAll("text")
+      .data(centuries)
+      .join('text')
+        .attr('x', min_x + 5)
+        .attr('y', (d, i) => (min_y + i * leafContainerHeight + 15))
+        .attr('fill', "gray")
+        .attr('style', 'font-size: 10px')
+        .text(d => d + '00s');
+
+  // calculate leaf constraints based on leaf container height and century
+  const leafConstraints = new Object();
+  centuries.forEach((c, i) => {
+    let localTop = leafContainerHeight * i;
+    leafConstraints[c] = {
+      top:  localTop,
+      bottom: localTop + leafContainerHeight
+    };
+  });
+
+  // draw a couple of lines to help gesture at tree-ness
+  let trunkWidth = 65;
+  // right side
+  background.append("path")
+    .attr("d", d3.line().curve(d3.curveNatural)([
```

Comment:
Could we be using these curved paths throughout the tree?

Reply:
yes! that is my thought for one of the next steps  — I want to draw branches over the network graph to connect the trunk to the leaves. Thinking we could implement a method that introduces some amount of variation between the starting point and ending point to make the lines look a little more natural and tree-like.